www/js/toolbar.js

```diff
@@ -107,34 +149,82 @@ const toolbarItemFactory = toolbarItem =>
     onclick: () => {
       const { databaseState } = global.core.search.$search.dataset;
       if (databaseState === 'loaded') {
-        toggleOverlayUI();
+        toggleOverlayUI(toolbarItem, true);
         if (!banisLoaded) {
           banidb.loadBanis().then(rows => {
             printBanis(rows);
             banisLoaded = !!rows;
           });
+
           analytics.trackEvent('banisLoaded', true);
         }
       }
     },
   });
 
-const closeOverlayUI = h(
-  'div.close-overlay-ui.overlay-ui.hidden',
-  {
-    onclick: toggleOverlayUI,
-  },
-  h('i.fa.fa-times'),
-);
+const printCeremonies = rows => {
+  rows.forEach(row => {
+    if (row.Token !== 'anand' && row.Token !== 'death') {
+      const $ceremony = h(
+        `div.ceremony-pane#${row.Token}`,
+        {
+          onclick: () => {
+            analytics.trackEvent('ceremony', row.Token);
+            global.core.search.loadCeremony(row.ID);
+            const currentCeremony = document.querySelector('div.ceremony-pane.active');
+            if (currentCeremony) {
+              currentCeremony.classList.remove('active');
+            }
+            document.querySelector(`div.ceremony-pane#${row.Token}`).classList.add('active');
+          },
+        },
+        navigatorHeaderFactory(row.Token, row.Gurmukhi, 'gurmukhi'),
+        h(
+          'div.ceremony-pane-content',
+          h(
+            `div.ceremony-pane-options#cpo-${row.Token}`,
+            switchFactory(
+              `${row.Token}-english-exp-switch`,
+              'English Explanations',
+              `${row.Token}-english-exp`,
+              () => {
+                const englishExpVal = store.getUserPref(`gurbani.ceremonies.${row.Token}-english`);
+                store.setUserPref(`gurbani.ceremonies.${row.Token}-english`, !englishExpVal);
+                global.platform.updateSettings();
+              },
+              store.getUserPref(`gurbani.ceremonies.${row.Token}-english`),
```

Comment:
think this evaluates to `undefined` the first time and makes the switch have a true value since
```switchFactory = (id, label, inputId, clickEvent, defaultValue = true) =>```
makes the toggle out of sync with the actual value.